inception/inception-api-annotation-experimental/src/main/ts/client/AnnotationExperienceAPIImpl.ts

```diff
@@ -0,0 +1,308 @@
+/*
+ * Licensed to the Technische Universität Darmstadt under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The Technische Universität Darmstadt
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Client, Stomp} from '@stomp/stompjs';
+import {AnnotationExperienceAPIVisualization} from "./visualization/AnnotationExperienceAPIVisualization";
+import {AnnotationExperienceAPIActionHandler} from "./actionhandling/AnnotationExperienceAPIActionHandler";
+import {ServerMessage} from "./util/ServerMessage";
+import {AnnotationExperienceAPI} from "./AnnotationExperienceAPI"
+import {Annotation} from "./util/Annotation";
+
+export class AnnotationExperienceAPIImpl implements AnnotationExperienceAPI {
+
+    //Websocket and stomp broker
+    stompClient: Client;
+    connected: boolean = false;
+
+    //States to remember by client
+    client: string;
+    projectID: string;
+
+    documentID: string;
+
+
+    //Text and annotations
+    text: String[];
+    annotations: Annotation[];
+    selectedAnnotation: Annotation;
+
+    //Viewport
+    viewport: number[][];
+
+    //Visualizer
+    readonly visualizer: AnnotationExperienceAPIVisualization;
+
+    //Actionhandler
+    readonly actionhandler: AnnotationExperienceAPIActionHandler;
+
+    constructor() {
+        this.connect();
+
+        //Visualizer
+        this.visualizer = new AnnotationExperienceAPIVisualization(this);
+
+        //ActionHandler
+        this.actionhandler = new AnnotationExperienceAPIActionHandler(this);
+
+        this.actionhandler.registerDefaultActionHandler();
+    }
+
+
+    //CREATE WEBSOCKET CONNECTION
+    connect() {
+        if (this.connected) {
+            console.log("You are already connected")
+            return;
+        }
+
+        this.stompClient = Stomp.over(function () {
+            return new WebSocket(localStorage.getItem("url"));
```

Comment:
Why do we need to make use of the local storage?

Reply:
I haven't found a better solution for typescript for the issue yet. As Ute uses vue's "module.export { props: .. }" for that. But i'll mark it for the final version of the API and fix it accordingly